src/compass/s1_geocode_slc.py

```diff
@@ -133,44 +147,74 @@ def run(cfg: GeoRunConfig):
                 if check_eap.phase_correction:
                     temp_slc_path_corrected = temp_slc_path.replace('_temp.vrt',
                                                                     '_corrected_temp.rdr')
-                    apply_eap_correction(b,
+                    apply_eap_correction(burst,
                                          temp_slc_path,
                                          temp_slc_path_corrected,
                                          check_eap)
 
                     # Replace the input burst if the correction is applied
                     temp_slc_path = temp_slc_path_corrected
 
-
-                rdr_burst_raster = isce3.io.Raster(temp_slc_path)
-
-                # Create HDF5 dataset for a given frequency and polarization
-                gslc_dataset = init_geocoded_dataset(grid_group, pol, geo_grid,
-                                                     'complex64',
-                                                     f'{pol} geocoded CSLC image')
-
-                # Construct the output raster directly from HDF5 dataset
-                geo_burst_raster = isce3.io.Raster(f"IH5:::ID={gslc_dataset.id.id}".encode("utf-8"),
-                                                   update=True)
+                # prepare input dataset as GDAL raster
+                rdr_dataset = gdal.Open(temp_slc_path, gdal.GA_ReadOnly)
+                rdr_datasets.append(rdr_dataset)
```

Comment:
Can you comment on what is going on here? So, we are opening GDAL rasters which correspond to the burst we want to process and stack them together in a list? Are we geocoding them together? I am confused because we are still under the `for` loop for the bursts

Reply:
This [first loop](https://github.com/opera-adt/COMPASS/blob/1a8db1f0a365070702560e6466c88fcdd28d243e/src/compass/s1_geocode_slc.py#L139-L167) puts radar and geo datasets into the same place on their respective lists so when they are iterated over independently [before](https://github.com/opera-adt/COMPASS/blob/1a8db1f0a365070702560e6466c88fcdd28d243e/src/compass/s1_geocode_slc.py#L182-L189)/[after](https://github.com/opera-adt/COMPASS/blob/1a8db1f0a365070702560e6466c88fcdd28d243e/src/compass/s1_geocode_slc.py#L204-L208) geocoding block processing the data gets written to the correct place in the HDF5. I will update comments to be more descriptive.